server/app/api.py

```diff
@@ -285,25 +290,31 @@ def statistics(self):
 
 
 class UserAPI(APIResource):
+    """The API resource for the User Object
 
-    """The API resource for the User Object"""
+    model- the class in models.py that this API is for.
+    key_type- The type of the id of this model.
+    """
     model = models.User
-    key_type = str
+    key_type = int
 
     methods = {
         'post': {
             'web_args': {
-                'first_name': Arg(str),
-                'last_name': Arg(str),
-                'email': Arg(str, required=True),
-                'login': Arg(str),
+                'email': KeyRepeatedArg(str, required=True),
+                'name': Arg(str),
             }
         },
-        'put': {
+        'add_email': {
+            'methods': set(['PUT']),
             'web_args': {
-                'first_name': Arg(str),
-                'last_name': Arg(str),
-                'login': Arg(str),
+                'email': Arg(str)
+            }
+        },
+        'delete_email': {
```

Comment:
Once all emails are deleted, a user will not be accessible. I think the standard solution is to prohibit deleting the last email. I'm adding a `_pre_put_hook` to the `User` model, but I'm not sure that's the best way to enforce this issue.